action.yml

```diff
@@ -62,30 +73,47 @@ runs:
             shutil.rmtree(venv_path)
         builder = EnvBuilder()
         builder.create(venv_path)
-        cibw_path = [path for path in builder.bin_path.glob("cibuildwheel*") if path.stem == "cibuildwheel"][0]
+        exposed_binaries = {"cibuildwheel"}
+        if "uv" in EXTRAS:
+            exposed_binaries.add("uv")
+        clean_bin_path = builder.bin_path.parent / f"{builder.bin_path.name}.clean"
+        clean_bin_path.mkdir()
+        for path in list(builder.bin_path.iterdir()):
+            if path.stem in exposed_binaries:
+                try:
+                    os.symlink(path, clean_bin_path / path.name)
+                except OSError:
+                    import shutil
+
+                    shutil.copy2(path, clean_bin_path / path.name)
+        full_path = f"{clean_bin_path}{os.pathsep}{os.environ['PATH']}"
         with open(os.environ["GITHUB_OUTPUT"], "at") as f:
-            f.write(f"cibw-path={cibw_path}\n")
+            f.write(f"updated-path={full_path}\n")
```

Comment:
I wonder, if we're gonna modify PATH within the scope of the action, is there a reason to not simplify a little and add the venv bin dir to the PATH instead?

```python
        builder = EnvBuilder()
        builder.create(venv_path)
        full_path = f"{builder.bin_path}{os.pathsep}{os.environ['PATH']}"
        with open(os.environ["GITHUB_OUTPUT"], "at") as f:
            f.write(f"updated-path={full_path}\n")
```

I believe this would still achieve the objective of #1745 - not contaminating the rest of the action context.

Reply:
Executables from transitive dependencies is a fair point. I'm not too concerned about those since any other method of invoking cibuildwheel is gonna have them on path (pipx, uvx etc) but I take the point that more isolated is better.